tests/test_binaries.py

```diff
@@ -1375,7 +1393,9 @@ def train_masked_lm(data_dir, arch, extra_flags=None):
             "no_c10d",
             "--num-workers",
             "0",
-        ]
+            "--dataset-impl",
+            "raw",
```

Comment:
why this change? Ideally we'd test with binarized data, since that's the more common usage

Reply:
The reason for switching the tests to raw is because of PermissionErrors raised due to strong references held on the binary memmap'd files in `MMapIndexedDataset`. The tests consistently end before the `FairseqTask` is gc'd which means the files are still "in use" causing huge failures. My thought was that your internal tests are likely 1) mmap-based 2) Linux-only which would also test this more comprehensively.

Reply:
[This commit](68d0af4a49e) adds the additional tests that are binarized but skips them on Windows. I added this to ensure no regressions are introduced on non-Windows platforms, but it would be nice to refactor these tests eventually. Probably not an immediate thing, though.

Reply:
ah, I think we don't need to switch to raw for this. If you switch to `--dataset-impl=cached` everywhere in tests (including preprocessing), I think it should load the dataset into memory without the mmap.